cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -995,6 +1008,19 @@ def __init__(
         super().__init__(resources, cognite_client)
         self.cursor = cursor
 
+    def extend(self, other: EdgeListWithCursor) -> None:  # type: ignore[override]
+        if not isinstance(other, type(self)):
+            raise ValueError("Unable to extend as the types do not match")
+        other_res_list = type(self)(other, other.cursor)  # See if we can accept the types
+        if set(self._id_to_item).isdisjoint(other_res_list._id_to_item):
```

Comment:
`_id_to_item` will always be empty, so this will always return `True`, need to use `_external_id_to_item`. Let's add unit tests for duplicates in extends.

Also 
```
set(self._external_id_to_item).isdisjoint(other_res_list._external_id_to_item)
```
is about 10 times slower than
```
self._external_id_to_item.keys().isdisjoint(other_res_list._external_id_to_item)
```

Reply:
Nice did not know that one, learned something new.